source/common/config/grpc_mux_impl.cc

```diff
@@ -206,5 +212,27 @@ void GrpcMuxImpl::handleEstablishmentFailure() {
   }
 }
 
+void GrpcMuxImpl::queueDiscoveryRequest(const std::string& queue_item) {
+  request_queue_.push(queue_item);
+  drainRequests();
+}
+
+void GrpcMuxImpl::clearRequestQueue() {
```

Comment:
Is there an opportunity to factor out the common queue management to another owns-a object or common base?

Reply:
If it's just `drainRequests` that is non-trivial, I agree it won't simplify much, so yeah, we can resolve this. Thanks for the rationale.

Reply:
Actually, good/bad news: I've realized that the existing "you can drop DiscoveryRequests whenever you want; later ones will take care of it" logic is incorrect for delta xDS. This queue logic had to be split out because the queue item was templated, but my fix (I will open a new PR for it) is going to entail using just string type_url as the queue item type for both SotW and delta. 

So, this ugliness will get cleaned up as a part of that.